src/wasm/module-decoder.cc

```diff
@@ -116,10 +116,17 @@ class ModuleDecoder : public Decoder {
                 {nullptr, 0, 0, 0, 0, 0, 0, false, false});
             WasmFunction* function = &module->functions->back();
             DecodeFunctionInModule(module, function, verify_functions);
-
-            if (ok() && verify_functions) {
-              if (!function->external)
+          }
+          if (ok() && verify_functions) {
```

Comment:
Ok, I had overlooked that. In that case can you change the loop above to pass verify_functions=false to avoid a redundant verification pass?